crates/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -5,6 +5,25 @@ use thiserror::Error;
 use crate::hir::resolution::errors::ResolverError;
 use crate::hir_def::expr::HirBinaryOp;
 use crate::hir_def::types::Type;
+use crate::Signedness;
+
+#[derive(Error, Debug, Clone, PartialEq, Eq)]
+pub enum Source {
+    #[error("Binary")]
+    Binary,
+    #[error("Assignment")]
+    Assignment,
+    #[error("Array")]
+    Array,
+    #[error("Array2")]
+    Array2,
+    #[error("Array2")]
+    String,
+    #[error("Comparison")]
+    Comparison,
+    #[error("BinOp")]
```

Comment:
Also, do these really need the Error annotations? Can we remove them?

Reply:
i added derive for trait Error due to error:
```
error[E0599]: the method `as_dyn_error` exists for reference `&type_check::errors::Source`, but its trait bounds were not satisfied
  --> crates/noirc_frontend/src/hir/type_check/errors.rs:37:72
   |
11 | pub enum Source {
   | ---------------
   | |
   | doesn't satisfy `type_check::errors::Source: AsDynError<'_>`
   | doesn't satisfy `type_check::errors::Source: StdError`
...
37 |     TypeMismatchWithSource { expected: Type, actual: Type, span: Span, source: Source },
   |                                                                        ^^^^^^ method cannot be called on `&type_check::errors::Source` due to unsatisfied trait bounds
   |
```